File: golang/fizz_buzz_cuckoo_clock.go
package kata

import (
	"fmt"
	"strings"
	"strconv"
)

func FizzBuzzCuckooClock(time string) string {
	var phrase string
	hour := time[0:2]
	minute := time[3:]

	h, err := strconv.Atoi(hour)
	if err != nil {
		fmt.Println("Error!")
	}
	m, err := strconv.Atoi(minute)
	if err != nil {
		fmt.Println("Error!")
	}

	switch {
	case m == 30:
		phrase = "Cuckoo"
	case m == 0 && h > 0 && h <= 12:
		phrase = strings.TrimSuffix((strings.Repeat("Cuckoo ", h)), " ")
	case m == 0 && h > 12:
		phrase = strings.TrimSuffix((strings.Repeat("Cuckoo ", (h-12))), " ")
	case m == 0 && h == 0:
		phrase = strings.TrimSuffix((strings.Repeat("Cuckoo ", 12)), " ")
	case m % 3 == 0 && m % 5 == 0 && m != 0:
		phrase = "Fizz Buzz"
	case m % 3 == 0 && m != 0:
		phrase = "Fizz"
	case m % 5 == 0 && m != 0:
		phrase = "Buzz"
	default:
		phrase = "tick"	
	}

	return phrase
}

File: js/fizz_buzz_cuckoo_clock.js
function fizzBuzzCuckooClock(time) {
	let h = time.slice(0, 2);
	let m = time.slice(3);
	let hour = parseInt(h, 10);
	let minute = parseInt(m, 10);

	let phrase;

	if(minute == 30) {
		phrase = "Cuckoo";
	} else if(minute == 0) {
		if(hour == 0) {
			phrase = "Cuckoo ".repeat(12).trim();
		} else if(hour > 0 && hour <= 12) {
			phrase = "Cuckoo ".repeat(hour).trim();
		} else {
			phrase = "Cuckoo ".repeat(hour - 12).trim();
		}
	} else if(minute % 3 == 0 && minute % 5 == 0) {
		phrase = "Fizz Buzz";
	} else if(minute % 3 == 0) {
		phrase = "Fizz";
	} else if(minute % 5 == 0) {
		phrase = "Buzz";
	} else {
		phrase = "tick"
	}

	return phrase
}